moira/db.py

```diff
@@ -1072,6 +1074,25 @@ def getEvents(self, trigger_id=None, start=0, end=100):
         defer.returnValue([anyjson.deserialize(e) for e in events])
 
     @defer.inlineCallbacks
+    @docstring_parameters(TARGETS)
+    def getTargets(self):
+        """
+        getTargets(self)
+
+        Returns all targets from db (variant = False)
+        or returns list targets in set TARGETS
+
+        :rtype: list
+        """
+        variant = False
+        if variant:
+            targets = yield self.rc.hkeys(TARGETS)
+            defer.returnValue(targets)
+        else:
+            targets = yield self.rc.keys("moira-metric-data*")
```

Comment:
This code is for autosuggest functionality on trigger editing screen. So, this piece of code will be called on each keypress of a user.

`KEYS` is a very inefficient operations, not recommended for production environments. You are basically giving users a weapon to overload the database.